albumentations/augmentations/transforms.py

```diff
@@ -3378,3 +3467,87 @@ def get_transform_init_args_names(self):
             "mask_fill_value",
             "random_offset",
         )
+
+
+class AugMix(ImageOnlyTransform):
+    """Augmentation from "AugMix: A Simple Data Processing Method to Improve Robustness and Uncertainty"
+    Please note that this augmentation performs normalization internally, so resulting image will be float type.
+
+    Args:
+        alpha (float): Probability coefficient for Beta and Dirichlet distributions. Default: (1.0).
+        width (int): Width of augmentation chain. Default: 3.
+        depth (int, tuple of ints): Depth of augmentation chain. If single int will be used provided number.
+            If tuple of depth will be generated in range `[depth[0], depth[1])`. Default: (3).
+        transforms (list of albumentation transforms): List of transforms from which augmentation will be sampled
+            on each step of AugMix procedure.
+        mean (float, list of floats, tuple of float): mean values for normalization. Default: (0.485, 0.456, 0.406).
+        std (float, list of floats, tuple of floats): std values for normalization. Default: (0.229, 0.224, 0.225).
+    Targets:
+        image
+
+    Image types:
+        uint8, float32 3-channel images only
+
+    Credit:
+        https://arxiv.org/pdf/1912.02781.pdf
+        https://github.com/google-research/augmix
+    """
+
+    def __init__(
+        self,
+        alpha=1.0,
+        width=3,
+        depth=3,
+        transforms=None,
+        mean=(0.485, 0.456, 0.406),
+        std=(0.229, 0.224, 0.225),
+        always_apply=False,
+        p=0.5,
+    ):
+        super().__init__(always_apply=always_apply, p=p)
+        self.alpha = alpha
+        self.width = width
+        self.depth = to_tuple(depth, low=1)
+        self.transforms = transforms
+
+        if self.transforms is None:
+            self.transforms = [
+                Autocontrast(),
+                Posterize(num_bits=(3, 4)),
+                ShiftScaleRotate(shift_limit=0, scale_limit=0, rotate_limit=5, border_mode=cv2.BORDER_CONSTANT),
+                Solarize(threshold=77),
+                RandomShear(shear_x=0.09, shear_y=0, border_mode=cv2.BORDER_CONSTANT),
+                RandomShear(shear_x=0, shear_y=0.09, border_mode=cv2.BORDER_CONSTANT),
+                ShiftScaleRotate(shift_limit=0.09, scale_limit=0, rotate_limit=0, border_mode=cv2.BORDER_CONSTANT),
+            ]
+
+        self.mean = mean
+        self.std = std
+
+    def apply(self, img, depth=3, random_state=None, **params):
+        return F.aug_mix(
+            img,
+            self.alpha,
+            self.width,
+            depth,
+            self.transforms,
+            self.mean,
+            self.std,
+            random_state=np.random.RandomState(random_state),
+        )
+
+    def get_params(self):
+        return {"depth": random.randint(self.depth[0], self.depth[1]), "random_state": random.randint(0, 10000)}
```

Comment:
Assigning a single `int` to `depth` here makes all augmentations of `width` have the same `depth` (this is in divergence to the reference implementation